src/main/php/com/amazon/aws/ServiceEndpoint.class.php

```diff
@@ -163,11 +168,11 @@ public function sign(string $target, int $expires= 3600, $time= null): string {
   }
 
   /**
-   * Sends a request and returns the response
+   * Opens a request and returns a `Transfer` instance for writing data to
    *
    * @throws io.IOException
    */
-  public function request(string $method, string $target, array $headers= [], $payload= null, $time= null): Response {
+  public function open(string $method, string $target, array $headers, $hash= null, $time= null): Transfer {
```

Comment:
We should test this with the special *UNSIGNED-PAYLOAD* value!

Reply:
...and these: *x-amz-content-sha256 must be UNSIGNED-PAYLOAD, STREAMING-UNSIGNED-PAYLOAD-TRAILER, STREAMING-AWS4-HMAC-SHA256-PAYLOAD, STREAMING-AWS4-HMAC-SHA256-PAYLOAD-TRAILER, STREAMING-AWS4-ECDSA-P256-SHA256-PAYLOAD, STREAMING-AWS4-ECDSA-P256-SHA256-PAYLOAD-TRAILER or a valid sha256 value*

Reply:
Works, see xp-forge/aws@0055a13